### PR TITLE
Fix latest "inspect" bug by adding manual Vercel Project Name option

### LINKED ISSUE #27: Deployment succeeds but action log says it failed
HI.

The setup in my **.yml** file is:

    name: Deployment

    on:
      push:
        branches:
      - master

    jobs:
     deploy:
     runs-on: ubuntu-latest
     steps:
      - uses: actions/checkout@v2
      - name: 'Set build id'
        id: build_id
        run: echo "::set-output name=id::$(date +%s)"

      - name: Deploy master branch
        if: github.ref == 'refs/heads/master'
        uses: amondnet/vercel-action@v19.0.1+2
        with:
          vercel-token: ${{ secrets.ZEIT_TOKEN }}
          vercel-args: '--prod -b BUILD=${{ steps.build_id.outputs.id }} -b BUILD_BRANCH=master -b ENVIRONMENT=production'
          vercel-org-id: ${{ secrets.ZEIT_ORG_ID }}
          vercel-project-id: ${{ secrets.ZEIT_PRODUCTION_PROJECT_ID }}

On push to master the action gets triggered but when it finishes action log says it failed (even tho code gets deployed and logs on vercel.com confirm succesful deployment)

Action logs:

    https://*****************.vercel.app- Queued

    - Queued

    - Building

    - Building

    Production: https://***************** [3m]
    Due to "builds" existing in your configuration file, the Build and Development Settings defined in your Project Settings will not apply. Learn More: https://vercel.link/unused-build-settings

    set preview-url output
    /usr/local/bin/npx vercel inspect https://*****************.vercel.app -t ***
    Vercel CLI 19.2.0

    Vercel CLI 19.2.0
    - Fetching deployment "https://*****************.vercel.app" in *****

    - Fetching deployment "https://*****************.vercel.app" in *****

    Error! Failed to find deployment "https://*****************.vercel.app" in *****

    Error! Failed to find deployment "https://*****************.vercel.app" in *****
    ##[error]The process '/usr/local/bin/npx' failed with exit code 1

Any ideas?

### PR DESCRIPTION
**Pull request by EvanLovely**
This would resolve #27 & #28 by sidestepping the need to even run `vercel inspect` as that is only needed to get the Vercel Project Name - this PR adds the ability to manually set `vercel-project-name` inside the GitHub Actions yaml config. Let me know if there's anything else needed for this to get in! Thanks!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -324,7 +325,10 @@ async function run() {
     core.warning('get preview-url error');
   }
 
-  const deploymentName = await vercelInspect(deploymentUrl);
+  let deploymentName = vercelProjectName;
+  if (!deploymentName) {
+    deploymentName = await vercelInspect(deploymentUrl);
+  }
```
Comment (simonhaenisch) [inline index.js:331]:
```suggestion
  const deploymentName = vercelProjectName ?? await vercelInspect(deploymentUrl);
```

Comment (EvanLovely) [reply index.js]: Added! Needed to include `()` around `await` though.

**Comment by klapec** [comment]
Is there anything blocking this pull request from being merged (aside from the checks, I guess?)
`vercel-action` is currently useless as it fails on the `vercel inspect` command.

**Comment by simonhaenisch** [comment]
Got this email yesterday

> Hello,

> The issue has been fixed for all new deployments. We are working to make it compatible with old deployments as well.

> Can you confirm?


> Best Regards,
> ~ Paulo De Mitri @ Vercel

**Pull request merged**